Attempt at explaining some of the om3 timesteps (#1235)
ACCESS-Hive/access-hive.github.io: docs/models/run_a_model/run_access-om3.md

!!! warning
    Note that the restart flag used here will only be applied if there is no restart directory in archive, and so does not have to be removed for subsequent submissions. See [Payu docs](https://payu.readthedocs.io/en/stable/config.html#miscellaneous) for further details.

### Changing model timesteps

> The model timesteps set how frequently the model recalculates it's internal state. 

Reviewer: ```suggestion
The model timesteps set how frequently the model recalculates its internal state. 
```

> Longer timesteps are more computational efficient, but the length is limited by numerical stability and the quality of the 

Reviewer: ```suggestion
Longer timesteps are more computationally efficient, but the length is limited by numerical stability and the quality of the 
```

> modeled result. In release configurations, the timesteps are set as long as practical to maintain model stability.

Reviewer: ```suggestion
modeled result. In released configurations, the timesteps are set as long as practical to maintain model stability.
```

Reviewer: I think I heard that's currently not the case for one of the panans? Bit of an edge case I guess, but is that right @dougiesquire?

Author: panans aren't released

If alternative new configurations are being used or developed, then a shorter timestep may be needed.

> There are several timesteps to be aware of for each active model component. The ocean (MOM6) has baratropic, baraclinic, thermodynamic, 

Reviewer: ```suggestion
There are several timesteps to be aware of for each active model component. The ocean (MOM6) has barotropic, baraclinic, thermodynamic, 
```

Reviewer: Somewhere I think it's worth noting that it's not necessary true split-explict time stepping (baro/baroclinic are used). In our case we have `SPLIT = True`:
https://github.com/ACCESS-NRI/access-om3-configs/blob/4429156d3bb1ad4e04f41be8a90329684365786d/docs/MOM_parameter_doc.all#L4

> There are several timesteps to be aware of for each active model component. The ocean (MOM6) has baratropic, baraclinic, thermodynamic, 

Reviewer: For this:
```
There are several timesteps to be aware of for each active model component. The ocean (MOM6) has baratropic, baraclinic, thermodynamic, tracer advection and remapping timesteps, whilst the sea ice (CICE) has thermodynamic and dynamic timesteps and there is also the interval 
which coupling occurs.
```

I would suggest
```
In OM3, there are several timesteps to be aware of for each model component: 

- ocean (MOM6) has barotropic, baroclinic, thermodynamic, tracer advection and remapping timesteps;
- sea ice (CICE) has thermodynamic and dynamic timesteps;
- coupling frequency (NUOPC).
```

tracer advection and remapping timesteps, whilst the sea ice (CICE) has thermodynamic and dynamic timesteps and there is also the interval 
which coupling occurs.

> This section introduces the baratropic, baraclinic, coupling and sea ice timesteps. Details on all the MOM6 time steps are in the 

Reviewer: ```suggestion
This section introduces the barotropic, baraclinic, coupling and sea ice timesteps. Details on all the MOM6 time steps are in the 
```

Reviewer: Also `baraclinic`

Reviewer: Would replace with...

The next two sections introduce the... 

MOM6 link could come earlier. Could also reference this video: https://www.youtube.com/watch?v=JKMwd8VXYcU

> [MOM6 documentation](https://mom6.readthedocs.io/en/main/api/generated/pages/Timestep_Overview.html).

Reviewer: I don't think it's helpful to link to docs that use different words to the ones used here. If you'd like to use those docs can I suggest we align with their language?

Also, I think @dougiesquire mentioned yesterday that there's a new additional time-step on the way?

Reviewer: (I imagine not yet in our released configs though?)

Author: which language is different specifically ? they use baroclinic / langrangian interchangably ( I think)

#### Ocean

The ocean timesteps can be set independently of the coupling interval.

> Reducing the baratropic timestep helps with stability issues in 2d dynamical ocean processes and is controlled through the parameter `DTBT` 

Reviewer: ```suggestion
Reducing the barotropic timestep helps with stability issues in 2d dynamical ocean processes and is controlled through the parameter `DTBT` 
```

Reviewer: I personally don't like the use of the word 2-d here (although the mom6 docs use it). I would prefer `depth-integrated` as it's really the shallow water equations that I think are best kept in mind.

Reviewer: Having said that, `depth-averaged` is used here so perhaps best to go with that: https://github.com/ACCESS-NRI/access-om3-configs/pull/1504/changes#r3575959453

> in `MOM_input`. This is typically set as a fraction of the maximum stable value but can be set in seconds. When `DTBT` is negative, the 

Reviewer: Would suggest including a link to the released config used in this doc to that file.

Reviewer: The config currently is: `release-MC_25km_jra_ryf`

so proposing 

https://github.com/ACCESS-NRI/access-om3-configs/blob/4429156d3bb1ad4e04f41be8a90329684365786d/MOM_input#L481

baratropic timestep is set as the fraction of the maximum stable value, which is calculated internally in MOM6. The timestep 
is recalculated at the frequency set through `DTBT_RESET_PERIOD`. Therefore reducing the magnitude of either `DTBT` or `DTBT_RESET_PERIOD`
should improve stability.

> See notes in `MOM_input` on the specifics of setting each parameter.

Reviewer: For me, this section is missing some key recommendations. A user I think would like to know what to do when the model crashes.

We could provide some general guidance here, e.g. try halving the existing time-step via modifying DT and the coupling interval, best to choose a new time-step that divides into a day etc.

Reviewer: It's not obvious to me why you've talked about BTBT first here? To my read it implies that one should modify that first which I don't think is the suggested approach?

> Reducing the baraclinc timestep can help with stability in 3d dynamical ocean processes. This is often set the same as the coupling timestep, or 

Reviewer: ```suggestion
Reducing the baroclinic timestep can help with stability in 3d dynamical ocean processes. This is often set the same as the coupling timestep, or 
```

can be set to a whole fraction of the coupling timestep. This is controled by the parameter `DT` in `MOM_input`, and the model will round the 
value set in `MOM_input` to the coupling interval or closest whole fraction of the coupling interval.

#### Sea Ice

> The sea ice thermodynamic timestep is equal to the coupling interval, can't be modified indepedently and should be numerically stable at all 

Reviewer: ```suggestion
The sea ice thermodynamic timestep is equal to the coupling interval, it cannot be modified independently and should be numerically stable at all 
```

Reviewer: Not sure what is meant by "should be numerically stable at all values"

values. The sea ice dynamics timestep is a whole fraction of the thermodynamic timestep, and the number of dynamic timesteps per 
thermodynamic timesteps is set through the `ndtd` paramater in `ice_in`. 

> 

Reviewer: > and the number of dynamic timesteps per thermodynamic timesteps is set through the `ndtd` paramater in `ice_in`. 

What is the reader to take from this? i.e. is it worth fiddling with the cice dynamic time-step?

Reviewer: Related: https://github.com/ACCESS-NRI/access-om3-configs/pull/1501/changes#r3575978427

#### Coupling

> The coupling interval is set in the seconds at the top of `nuopc.runseq` file :

Reviewer: ```suggestion
The coupling interval is set in seconds at the top of `nuopc.runseq` file :
```

```
runSeq::
  @300        # <-- coupling time-step in seconds
```

> This can be reduced when instabilities appear related to coupled fields, or it's unclear where the instability is occuring. 

Reviewer: ```suggestion
This can be reduced when instabilities appear related to coupled fields, or it is unclear where the instability is occurring. 
```

> There's some further discussion of the coupling time step in the 

Reviewer: ```suggestion
There is some further discussion of the coupling time step in the 
```
